homeassistant/components/sensor/rmvtransport.py

```diff
@@ -0,0 +1,210 @@
+        return self._name
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._state != STATE_UNKNOWN
+
+    @property
+    def state(self):
+        """Return the next departure time."""
+        self._state = self.data.departures[0].get('minutes', None)
+        return self._state
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        result = {}
+        try:
+            result = {
+                'next_departures': [val for val in self.data.departures[1:]],
+                'direction': self.data.departures[0].get('direction'),
+                'line': self.data.departures[0].get('line'),
+                'minutes': self.data.departures[0].get('minutes'),
+                'departure_time':
+                    self.data.departures[0].get('departure_time'),
+                'product': self.data.departures[0].get('product'),
+            }
+        except IndexError:
+            pass
+        return result
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return "min"
+
+    def update(self):
+        """Get the latest data and update the state."""
+        self.data.update()
+        if not self.data.departures:
+            self._state = None
+            self._icon = ICONS[None]
+            return
+        if self._name == DEFAULT_NAME:
+            self._name = self.data.station
+        self._station = self.data.station
+        self._state = self.data.departures[0].get('minutes')
+        self._state = self.data.departures[0].get('departure_time')
+        self._icon = ICONS[self.data.departures[0].get('product')]
+        return
+
+
+class RMVDepartureData:
+    """Pull data from the opendata.rmv.de web page."""
+
+    def __init__(self, station_id, destinations, directions,
+                 lines, products, timeoffset, maxjourneys):
+        """Initialize the sensor."""
+        import RMVtransport
+        self.station = None
+        self._station_id = station_id
+        self._destinations = destinations
+        self._directions = directions
+        self._lines = lines
+        self._products = products
+        self._timeoffset = timeoffset
+        self._maxjourneys = maxjourneys
+        self.rmv = RMVtransport.RMVtransport()
+        self.departures = []
+
+    def update(self):
+        """Update the connection data."""
+        try:
+            _data = self.rmv.get_departures(self._station_id,
+                                            products=self._products,
+                                            maxJourneys=50)
+        except ValueError:
+            self.departures = {}
+            _LOGGER.warning("Returned data not understood")
+            return
+        self.station = _data.get('station', None)
```

Comment:
`None` is the default value returned when the key is missing in the dict when using `dict.get`.